### PR TITLE
[Plugin] stable diffusion: add Metal support

### PR DESCRIPTION
**Pull request by dm4**
- Add Metal support by `WASMEDGE_PLUGIN_STABLEDIFFUSION_METAL`
- Build and upload plugin with Metal support on CI. Pack with related files (`.metal`, `.h` files).

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review agent](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by dm4** [comment]
Looks like the GitHub macOS arm64 runner does not support some instructions:

```
ggml_metal_graph_compute_block_invoke: error: unsupported op 'MUL_MAT'
GGML_ASSERT: /Users/runner/work/WasmEdge/WasmEdge/build/_deps/stable-diffusion-src/ggml/src/ggml-metal.m:918: !"unsupported op"
```

I will temporarily remove the tests on metal build.

Related issues:
- https://github.com/ggerganov/llama.cpp/issues/4998
- https://github.com/ggerganov/llama.cpp/pull/4794

**Comment by dm4** [comment]
Just verified that the built asset is compiled with Metal support:

```
> unzip WasmEdge-plugin-wasmedge_stablediffusion-metal-0.14.1-rc.2-3-ga0a3c117-darwin_23_arm64.tar.gz.zip
Archive:  WasmEdge-plugin-wasmedge_stablediffusion-metal-0.14.1-rc.2-3-ga0a3c117-darwin_23_arm64.tar.gz.zip
  inflating: plugin_wasmedge_stablediffusion.tar.gz

> tar zxvf plugin_wasmedge_stablediffusion.tar.gz
x libwasmedgePluginWasmEdgeStableDiffusion.dylib
x ggml-metal.metal
x ggml-common.h

> otool -L libwasmedgePluginWasmEdgeStableDiffusion.dylib
libwasmedgePluginWasmEdgeStableDiffusion.dylib:
        @rpath/libwasmedgePluginWasmEdgeStableDiffusion.dylib (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libwasmedge.0.dylib (compatibility version 0.0.0, current version 0.1.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1345.120.2)
        /System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate (compatibility version 1.0.0, current version 4.0.0)
        /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 2503.1.0)
        /System/Library/Frameworks/Metal.framework/Versions/A/Metal (compatibility version 1.0.0, current version 343.19.0)
        /System/Library/Frameworks/MetalKit.framework/Versions/A/MetalKit (compatibility version 1.0.0, current version 165.2.0)
        /System/Library/Frameworks/MetalPerformanceShaders.framework/Versions/A/MetalPerformanceShaders (compatibility version 1.0.0, current version 127.5.2)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1700.255.5)
        /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 2503.1.0)
        /usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
```

**Review by 0yi0 — APPROVED** [review]
lgtm

File: .github/workflows/build-extensions.yml [hunk 1]
```diff
@@ -676,3 +676,72 @@ jobs:
...
+      output_bins: libwasmedgePluginWasmEdgeStableDiffusion.dylib
+    needs: [ get_version ]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Build and install dependencies
+        run: |
+          eval $(/opt/homebrew/bin/brew shellenv)
+          brew install llvm@16 ninja cmake openssl opencv rust ffmpeg@6
```
Comment (hydai) [inline .github/workflows/build-extensions.yml:708]: I don't think we need `opencv`, `openssl`, and `rust` for the stable diffusion plugin.

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/build-extensions.yml (hunk 1 above)
Comment (dm4) [reply .github/workflows/build-extensions.yml]: removed

**Review by hydai — APPROVED** [review]
LGTM

**Pull request merged**